File: day9-remove-duplicates.js
/*
26. Remove Duplicates from Sorted Array (Easy)
(See CPP file for full problem description)
*/

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let officer = 0;
  let result = 1;
  let cm = 1;
  while (cm < nums.length) {
    if (nums[cm] === nums[cm - 1]) {
      cm++;
      continue;
    }
    nums[officer + 1] = nums[cm];
    result += 1;
    officer++;
    cm++;
  }
  return result;
};

File: Day09/day9-remove-duplicates.java
/*
26. Remove Duplicates from Sorted Array (Easy)
(See JS/CPP file for full problem description)
*/

class Solution {
    public int removeDuplicates(int[] nums) {
        int officer = 0;
        int result = 1;
        int cm = 1;
        while(cm < nums.length){
            if(nums[cm] == nums[cm-1]){
                cm++;
                continue;
            }
            nums[officer+1] = nums[cm];
            result += 1;
            officer++;
            cm++;
        }
        return result;
    }
}